File: ios/Objective-C/HelloWorldObjC/HelloWorldObjC/Results/DLRResultTableViewCell.m
//
//  DLRResultTableViewCell.m
//  HelloWorldObjC
//
//  Created by Dynamsoft's mac on 2022/9/9.
//

#import "DLRResultTableViewCell.h"

@interface DLRResultTableViewCell ()

@property (nonatomic, strong) UILabel *resultLabel;

@end

@implementation DLRResultTableViewCell

+ (CGFloat)cellHeightWithString:(NSString *)resultString {
    return [[DynamsoftToolsManager manager] calculateHeightWithText:resultString font:[UIFont systemFontOfSize:KDLRResultTextFont] AndComponentWidth:KDLRResultTextWidth] + 10;
}

- (instancetype)initWithStyle:(UITableViewCellStyle)style reuseIdentifier:(NSString *)reuseIdentifier {
    self = [super initWithStyle:style reuseIdentifier:reuseIdentifier];
    if (self) {
        [self setupUI];
    }
    return self;
}

- (void)setupUI {
    self.selectionStyle = UITableViewCellSelectionStyleNone;
    self.backgroundColor = [UIColor clearColor];
    
    [self addSubview:self.resultLabel];
}

- (void)updateUIWithString:(NSString *)resultString {
    self.resultLabel.text = resultString;
    self.resultLabel.height = [[DynamsoftToolsManager manager] calculateHeightWithText:resultString font:[UIFont systemFontOfSize:KDLRResultTextFont] AndComponentWidth:KDLRResultTextWidth];
}

// MARK: - Lazy
- (UILabel *)resultLabel {
    if (!_resultLabel) {
        _resultLabel = [[UILabel alloc] initWithFrame:CGRectMake(kComponentLeftMargin, 0, KDLRResultTextWidth, 0)];
        _resultLabel.textColor = [UIColor whiteColor];
        _resultLabel.font = [UIFont systemFontOfSize:KDLRResultTextFont];
        _resultLabel.numberOfLines = 0;
    }
    return _resultLabel;
}

@end

File: ios/Swift/HelloWorldSwift/HelloWorldSwift/Results/DLRResultTableViewCell.swift
//
//  DLRResultTableViewCell.swift
//  HelloWorldSwift
//
//  Created by Dynamsoft's mac on 2022/9/9.
//

import UIKit

class DLRResultTableViewCell: UITableViewCell {

    lazy var resultLabel: UILabel = {
        let resultLabel = UILabel.init(frame: CGRect.init(x: kComponentLeftMargin, y: 0, width: KDLRResultTextWidth, height: 0))
        resultLabel.textColor = .white
        resultLabel.font = UIFont.systemFont(ofSize: KDLRResultTextFont)
        resultLabel.numberOfLines = 0
        return resultLabel
    }()
    
    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
        super.init(style: style, reuseIdentifier: reuseIdentifier)
        setupUI()
    }
    
    required init?(coder: NSCoder) {
        super.init(coder: coder)
    }
    
    private func setupUI() -> Void {
        self.selectionStyle = .none
        self.backgroundColor = .clear
        self.contentView.addSubview(resultLabel)
    }
    
    class func getcellHeight(WithString resultString: String) -> CGFloat {
        return DynamsoftToolsManager.shared.calculateHeight(withText: resultString, font: UIFont.systemFont(ofSize: KDLRResultTextFont), componentWidth: KDLRResultTextWidth) + 10
    }
    
    func updateUI(withString resultString: String) -> Void {
        resultLabel.text = resultString
        resultLabel.height = DynamsoftToolsManager.shared.calculateHeight(withText: resultString, font: UIFont.systemFont(ofSize: KDLRResultTextFont), componentWidth: KDLRResultTextWidth)
    }
    
}
